incubator/validator-core/src/main/java/io/aklivity/zilla/runtime/validator/core/IntegerValidatorFactory.java

```diff
@@ -39,9 +39,32 @@ public URL schema()
     }
 
     @Override
-    public Validator create(
+    public ValueValidator createValueReader(
+        ValidatorConfig config,
+        LongFunction<CatalogHandler> supplyCatalog)
+    {
+        return new IntegerValidator(IntegerValidatorConfig.class.cast(config));
+    }
+
+    @Override
+    public ValueValidator createValueWriter(
+        ValidatorConfig config,
+        LongFunction<CatalogHandler> supplyCatalog)
+    {
+        return new IntegerValidator(IntegerValidatorConfig.class.cast(config));
+    }
+
+    @Override
+    public FragmentValidator createFragmentReader(
+        ValidatorConfig config,
+        LongFunction<CatalogHandler> supplyCatalog)
+    {
+        return new IntegerValidator(IntegerValidatorConfig.class.cast(config));
+    }
+
+    @Override
+    public FragmentValidator createFragmentWriter(
```

Comment:
Follow same pattern with private `createReader` and `createWriter` methods, delegating from public methods accordingly.